--- a/jenkins_jobs/parser.py
+++ b/jenkins_jobs/parser.py
@@ -250,6 +250,9 @@ class YamlParser(object):
             seen.add(job['name'])
 
     def expandYamlForTemplateJob(self, project, template, jobs_glob=None):
+        if 'job-name' in template:
+            logger.debug("Setting job name to {0}".format(template['job-name']))
+            template['name'] = template['job-name']
         dimensions = []
         template_name = template['name']
         # reject keys that are not useful during yaml expansion
